uikit/poidetail/src/main/java/com/sygic/ui/view/poidetail/PoiDetailBottomDialogFragment.kt

```diff
@@ -0,0 +1,117 @@
+
+        @JvmStatic
+        fun newInstance(poiData: PoiData): PoiDetailBottomDialogFragment = PoiDetailBottomDialogFragment().apply {
+            arguments = Bundle().apply {
+                putParcelable(POI_DATA, poiData)
+            }
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        preferencesManager = PreferencesManager(requireContext())
+
+        viewModel = ViewModelProviders.of(
+            this,
+            PoiDetailInternalViewModel.ViewModelFactory(
+                arguments?.getParcelable(POI_DATA)!!,
+                preferencesManager
+            )
+        )[PoiDetailInternalViewModel::class.java].apply {
+            this.setListener(listener)
+            listener = null
+
+            this.expandObservable.observe(this@PoiDetailBottomDialogFragment, Observer<Any> { expandBottomSheet() })
+            this.collapseObservable.observe(this@PoiDetailBottomDialogFragment, Observer<Any> { collapseBottomSheet() })
+            this.webUrlClickObservable.observe(this@PoiDetailBottomDialogFragment, Observer<String> { context?.openUrl(it) })
+            this.emailClickObservable.observe(this@PoiDetailBottomDialogFragment, Observer<String> { context?.openEmail(it) })
+            this.phoneNumberClickObservable.observe(this@PoiDetailBottomDialogFragment, Observer<String> { context?.openPhone(it) })
+            this.coordinatesClickObservable.observe(this@PoiDetailBottomDialogFragment, Observer<String> { context?.copyToClipboard(it) })
+        }
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        return BottomSheetDialog(
+            requireContext(),
+            theme,
+            resources.getDimensionPixelSize(R.dimen.defaultPeekHeight),
+            if (preferencesManager.showcaseAllowed) SHOWCASE_BEHAVIOR_STATE else DEFAULT_BEHAVIOR_STATE
+        )
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        binding = LayoutPoiDetailInternalBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.poiDetailInternalViewModel = viewModel
+        binding.setLifecycleOwner(this)
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        viewModel?.let { (dialog as BottomSheetDialog).behavior?.addStateListener(it) }
+    }
+
+    private fun expandBottomSheet() {
+        (dialog as BottomSheetDialog).behavior?.state = BottomSheetBehavior.STATE_EXPANDED
+    }
+
+    private fun collapseBottomSheet() {
+        (dialog as BottomSheetDialog).behavior?.state = BottomSheetBehavior.STATE_COLLAPSED
+    }
+
+    fun setListener(listener: Listener) {
+        viewModel?.setListener(listener) ?: run { this.listener = listener }
```

Comment:
ako sme to s tym listenerom nakoniec vymysleli? To ze ten VM ho udrzi cez rotacie nie je moc safe a on sam si ho clearuje az v onCleared... spoliehame sa na to, ze po rotacii sa "presetuje"?

Reply:
jj, při každé rotaci se setne novým nebo null, však víš jaké to bylo s tím bylo těžké, udělat to alespoň takto :(